### ISSUE TITLE
Test errors are emitted outside the Flow in `MapTestNetworkTransport`

### ISSUE THREAD

**Issue by TarasZubreiFiverr** [apollographql/apollo-kotlin#4944]
### Version

3.7.3

### Summary

In case you `registerTestNetworkError()` with `MapTestNetworkTransport` it throws exception eagerly [outside of `flow {}`](https://github.com/apollographql/apollo-kotlin/blob/78aaa0defc3b73ddd29d571ba33b3123282b7b88/libraries/apollo-testing-support/src/commonMain/kotlin/com/apollographql/apollo3/testing/TestNetworkTransport.kt#L67).
I've checked commit history and you already fixed the same bug in `QueueTestNetworkTransport`, but for some reason `MapTestNetworkTransport` wasn't patched:
* #4224

I know that there's an ongoing refactoring of error handling for 4.x (in #4711), but until it's out It would be great to fix this bug in 3.x

### Steps to reproduce the behavior

```kotlin
val apolloClient = ApolloClient.Builder()
    .networkTransport(MapTestNetworkTransport())
    .build()

apolloClient.registerTestNetworkError(DummyQuery())

apolloClient.query(DummyQuery())
    .toFlow() // this line will throw the `ApolloNetworkException` hence
    .retry(1) // nothing will be retried
    .catch { /* nothing will be caught here */ }
```

### Logs

```
java.util.concurrent.CompletionException: com.apollographql.apollo3.exception.ApolloNetworkException: Network error queued in QueueTestNetworkTransport
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at kotlinx.coroutines.future.CompletableFutureCoroutine.onCancelled(Future.kt:66)
	at kotlinx.coroutines.AbstractCoroutine.onCompletionInternal(AbstractCoroutine.kt:91)
	at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:236)
	at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.kt:909)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:866)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:831)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:100)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: com.apollographql.apollo3.exception.ApolloNetworkException: Network error queued in QueueTestNetworkTransport
	at com.apollographql.apollo3.testing.MapTestNetworkTransport.execute(TestNetworkTransport.kt:67)
	at com.apollographql.apollo3.interceptor.NetworkInterceptor.intercept(NetworkInterceptor.kt:22)
	at com.apollographql.apollo3.interceptor.DefaultInterceptorChain.proceed(ApolloInterceptor.kt:23)
	at com.apollographql.apollo3.ApolloClient.executeAsFlow(ApolloClient.kt:178)
	at com.apollographql.apollo3.ApolloCall.toFlow(ApolloCall.kt:97)
```

**Comment by BoD** [comment]
Thanks for reporting this! Would you be interested in opening a PR with the fix? If not no worries, I can do it!

**Comment by martinbonnin** [comment]
Fix is available in 3.8.2
